File: mpcop/src/main/java/com/mpcopenplatform/controller/GeneralMPCOPException.java
package com.mpcopenplatform.controller;

/**
 * The {@link GeneralMPCOPException} is used for error handling inside the MPCOP
 */
public class GeneralMPCOPException extends Exception {
    public GeneralMPCOPException(String message) {
        super(message);
    }

    public GeneralMPCOPException(String message, Throwable cause) {
        super(message, cause);
    }

    public GeneralMPCOPException(Throwable cause) {
        super(cause);
    }
}

File: mpcop/src/main/kotlin/cz/muni/fi/mpcop/GeneralMPCOPException.kt
package cz.muni.fi.mpcop

/**
 * The [GeneralMPCOPException] is used for error handling inside MPCOP
 */
class GeneralMPCOPException: Exception {
    constructor() : super()
    constructor(message: String) : super(message)
    constructor(message: String, cause: Throwable) : super(message, cause)
    constructor(cause: Throwable) : super(cause)
}
